### PR TITLE
Port forward-line to Rust

### PR DESCRIPTION
**Pull request by dcluna**
This should take care of #406.

### CODE REVIEW

**Comment by shaleh** [comment]
Checkout PR #517 to catch when you forget to run rustfmt. Or look at `rust_src/admin/rustfmt-helper` for how to have emacs call rustfmt whenever you save.

**Comment by dcluna** [comment]
@shaleh Thanks, that hook is very helpful. One thing though, after rebasing and adding your hook to my [Overcommit](https://github.com/brigade/overcommit) config, I'm not able to commit due to formatting differences in the following files: `rust_src/src/{base64.rs,crypto/mod.rs,floatfns.rs,fns.rs,hashtable.rs,lisp.rs,lists.rs,minibuf.rs,multibyte.rs,numbers.rs,obarray.rs,strings.rs,symbols.rs,windows.rs}`

Does the same happen to you? Should I make a different PR to address these formatting issues in the `rust_src`, or add an extra commit in this one?

**Comment by shaleh** [comment]
@dcluna ensure you are running the same version as the rest of us. You want rustfmt-nightly, 0.2.15. Checking the .travis.yml will always point you to the right version.

File: rust_src/src/cmds.rs [hunk 1]
```diff
@@ -154,6 +155,50 @@ pub fn end_of_line(n: LispObject) -> LispObject {
...
+
+    if shortage > 0
+        && (count <= 0
+            || (cur_buf.zv() > cur_buf.begv && cur_buf.pt() != opoint
+                && (cur_buf.fetch_byte(cur_buf.pt_byte - 1) != '\n' as u8)))
+    {
+        shortage -= 1
+    };
+
+    return LispObject::from_fixnum(if count <= 0 { -shortage } else { shortage } as EmacsInt);
```
Comment (shaleh) [inline rust_src/src/cmds.rs]: no return here

**Pull request merged**